app/src/main/java/com/testtask/apptesttask/ui/global/CharactersAdapter.kt

```diff
@@ -34,7 +30,57 @@ class CharactersAdapter constructor(
 
     override fun onBindViewHolder(holder: CharactersHolder, positon: Int) {
         val character = characters[positon]
-        holder.bind(character.apiCharacter)
-        holder.updateStar(character.favorite)
+        holder.bind(character)
+    }
+
+    override fun onBindViewHolder(
+        holder: CharactersHolder,
+        position: Int,
+        payloads: MutableList<Any>
+    ) {
+        super.onBindViewHolder(holder, position, payloads)
+
+        val character = characters[position]
```

Comment:
You don't need two versions of `onBindViewHolder`. If `payloads` are empty, then use `holder.bind(character)`, else `holder.updateStar(character.favorite!!)`.